server/vulnerabilities/msrc/parsed/product.go

```diff
@@ -66,7 +66,53 @@ func (p Products) GetMatchForOS(ctx context.Context, os fleet.OperatingSystem) (
 }
 
 func NewProductFromFullName(fullName string) Product {
-	return Product(fullName)
+	// If the full name includes a version, return it as-is.
+	p := Product(fullName)
+	if p.HasDisplayVersion() {
+		return p
+	}
+
+	// Several Windows products listed in MSRC bulletins don't include the OS version number.
+	// We need this to match the product with a host's OS, so we'll add them here.
+	versionString := ""
+	switch {
+	case strings.Contains(fullName, "Windows Server 2022"):
+		versionString = "21H2"
+
+	case strings.Contains(fullName, "Windows Server 2016"):
+		versionString = "1607"
+
+	case strings.Contains(fullName, "Windows Server 2019"):
+		versionString = "1809"
+
+	case strings.Contains(fullName, "Windows 8.1"):
+		versionString = "6.3 / NT 6.3"
```

Comment:
I've seen it both ways (6.3 and NT 6.3) and it's hard to say what the devices report, so putting both to be safe.